broker/pkg/coordinator/grpcserver.go

```diff
@@ -244,11 +245,14 @@ func (s *GRPCServer) TimelineDelete(stream grpc.Broker_TimelineDeleteServer) err
 			}
 			return err
 		}
+		consumer, err := s.listeners.GetConsumer(stream.Context(), string(req.SessionID()))
```

Comment:
You do not know if a Consumer or a producer called the delete. We cannot do this here.